actor-tests/src/test/scala/org/apache/pekko/actor/ActorSystemSpec.scala

```diff
@@ -125,23 +123,17 @@ object ActorSystemSpec {
 }
 
 @nowarn
-class ActorSystemSpec extends PekkoSpec(ActorSystemSpec.config) with ImplicitSender {
+class ActorSystemSpec extends PekkoSpec(ActorSystemSpec.config) with ImplicitSender with ScalaCheckPropertyChecks {
 
   import ActorSystemSpec.FastActor
 
   "An ActorSystem" must {
 
     "reject invalid names" in {
-      for (n <- Seq(
-          "-hallowelt",
-          "_hallowelt",
-          "hallo*welt",
-          "hallo@welt",
-          "hallo#welt",
-          "hallo$welt",
-          "hallo%welt",
-          "hallo/welt")) intercept[IllegalArgumentException] {
-        ActorSystem(n)
+      forAll { (name: String) =>
+        whenever(!name.matches("""^[a-zA-Z0-9][a-zA-Z0-9-_]*$""")) {
+          intercept[IllegalArgumentException](ActorSystem(name))
+        }
```

Comment:
I prefer the old way - you can actually see what is being tested. With the change, I have to hope that use cases are tested.

Reply:
Then there will be infinite cases to add

Reply:
Makes sense. Actually I felt like adding one more common case "hallo welt".

Reply:
keep the old test and add an extra new test that uses the property based approach

Reply:
updated, PTAL